analyzer/src/main/java/nl/tudelft/IndexerReader.java

```diff
@@ -18,36 +18,41 @@ public class IndexerReader {
         this.db = db;
     }
 
-    public List<String[]> indexerReader(String indexFile) throws IOException, SQLException {
+    public void indexerReader(String indexFile) throws IOException, SQLException {
         boolean checked = false;
         File file = new File(indexFile);
-        FileInputStream fileInputStream = new FileInputStream(file);
-        ChunkReader reader = new ChunkReader("index", fileInputStream);
-        Iterator<Map<String, String>> itr = reader.iterator();
         List<String[]> indexInfo = new ArrayList<>();
-        int i = 0;
-        while (itr.hasNext() && i < 2048) {
-            Map<String, String> chunk = itr.next();
-            if(chunk.get("u") != null) {
-                String[] tokens = (chunk.get("u").split("\\|"));
-                String[] arti = (chunk.get("i").split("\\|"));
-                String [] newList = new String[5];
-                System.arraycopy(tokens, 0, newList, 0, 3);
-                String epochDate = chunk.get("m");
-                newList[3] = epochDate;
-                newList[4] = arti[arti.length - 1];
-                indexInfo.add(newList);
-                i++;
-                if(i == 2048) {
-                    putInDatabase(indexInfo, checked);
-                    checked = true;
-                    i = 0;
-                    indexInfo = new ArrayList<>();
+        try (FileInputStream fileInputStream = new FileInputStream(file);
+             ChunkReader reader = new ChunkReader("index", fileInputStream)) {
+            Iterator<Map<String, String>> itr = reader.iterator();
+            int i = 0;
+            while (itr.hasNext() && i < 2048) {
+                Map<String, String> chunk = itr.next();
+                if (chunk.get("u") != null) {
+                    String[] tokens = (chunk.get("u").split("\\|"));
+                    String[] arti = (chunk.get("i").split("\\|"));
+                    String[] newList = new String[5];
+                    System.arraycopy(tokens, 0, newList, 0, 3);
+                    String epochDate = chunk.get("m");
+                    newList[3] = epochDate;
+                    newList[4] = arti[arti.length - 1];
+                    //TODO - DELETE BEFORE MERGING IN DEV
+                    if(newList[1].equals("yamcs-api")) {
+                        System.out.println("kkkk");
```

Comment:
huh?

Reply:
That statement has been put there intentionally, for debugging and making review easy for particular packages. It shall be removed once approved. That is why the TODO.